helper/convert/value.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package convert
+
+import (
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+func PrimitiveTfValue(in cty.Value) tftypes.Value {
+	if in.IsNull() {
+		return emptyTfValue(ToTfType(in.Type()))
+	}
+
+	var val tftypes.Value
+	switch in.Type() {
```

Comment:
I'm not exactly sure what the best way to handle this scenario, but what if the type is `cty.DynamicPseudoType`.

I'm fairly certain the only way that can be possible is if `in.IsUnknown()` or `in.IsNull()` are true, so perhaps a panic would be fine? 🤔